### ISSUE TITLE
Error: java.lang.NoClassDefFoundError: org/apache/spark/sql/delta/files/TahoeLogFileIndex

### ISSUE THREAD

**Issue by arleycg** [microsoft/hyperspace#356]
I'm executing Hyperspace v.0.4.0 on a Data Databricks Cluster 6.4 (includes Apache Spark 2.4.5, Scala 2.11) using "com.microsoft.hyperspace:hyperspace-core_2.11:0.4.0"

When I create an index using a dataframe from parquet source it works 100%, but when trying to create an index using a dataframe from a Delta source I'm getting the following error even after configured spark.conf.set:

java.lang.NoClassDefFoundError: org/apache/spark/sql/delta/files/TahoeLogFileIndex

Does anyone know what's happening?

**Comment by andrei-ionescu** [comment]
Yes. Databricks Runtime uses a special built version of Apache Spark in which the Delta Lake implementation is totally different that the OSS Delta.

This is the reason that the `TahoeLogFileIndex` is not found when running Hyperspace on DBR. It may not be even use in their own Delta implementation.

I'm sorry for this bad news but to use Delta with Hyperspace you need to switch to OSS Spark.

**Comment by Priyanshu-Mittal** [comment]
Are there any active plans for resolving this issue ?

**Comment by panish-git** [comment]
Similar issue with OSS Spark -> spark 3.0.1, Scala - 2.12, Delta - 2.12_0.7.0 using Hyperspace version 3.0 (for spark 3.0 , built from source). Please update.
Py4JJavaError: An error occurred while calling o130.createIndex.
: java.lang.NoClassDefFoundError: org/apache/spark/sql/delta/files/TahoeLogFileIndex
	at com.microsoft.hyperspace.index.sources.delta.DeltaLakeFileBasedSource.isSupportedRelation(DeltaLakeFileBasedSource.scala:49)

**Comment by sezruby** [comment]
@panish-git Could you check if delta dependency is added properly and also try to test using delta 0.8.0?
